src/integration-test/java/com/mongodb/jdbc/integration/testharness/IntegrationTestUtils.java

```diff
@@ -380,8 +396,9 @@ private static void validateRowCount(
 
     private static boolean compareRow(List<Object> expectedRow, ResultSet actualRow)
             throws SQLException {
+        boolean result = true;
         ResultSetMetaData rsMetadata = actualRow.getMetaData();
-        assertEquals(expectedRow.size(), rsMetadata.getColumnCount());
+        assertEquals(expectedRow.size(), rsMetadata.getColumnCount(), "Either the yml test specification is missing entries for 'expected_result' or the data in the DB is not matching the test");
```

Comment:
Nit: This assert is regarding a column count mismatch between the expected row and the metadata.  
I'd reword it to something related to the column count, maybe: "Column count mismatch between 'expected_result' in the yml test specification and ResultSetMetaData in the DB"